dotcom-rendering/src/lib/ArticleRenderer.tsx

```diff
@@ -85,6 +87,7 @@ export const ArticleRenderer = ({
 				editionId={editionId}
 				totalElements={length}
 				isSectionedMiniProfilesArticle={isSectionedMiniProfilesArticle}
+				shouldHideAds={shouldHideAds}
```

Comment:
I haven't checked exhaustively but if the decision to render ads is always dependent on checking all of `isAdFreeUser`, `isSensitive` & `shouldHideAds` booleans, then could this check be lifted higher up and we pass a single boolean such as `canRenderAds` down?